### DISCUSSION TITLE
Official FAST Fluent Blazor components

### DISCUSSION THREAD

**Discussion by limefrogyank** [BlazorFluentUI/BlazorFluentUI#309] [General]
https://github.com/microsoft/fast-blazor

So they're official now.  These are web component versions of the basic UI elements done using fluent styling.  

1.  They are not exactly the same style-wise as the react components used by the office team.
2.  They are limited to basic components.  

I haven't been able to do much lately with this repo given that I have too many other things going on, but I got the notification for this and have always been impressed by how nicely these components work.  They use FAST, but that is just a simple wrapper around web components.  If you haven't tried web components, it's basically a way to package html into a new component with its own tag.  

The limitations of FAST and web components in general is that the attributes you can modify them with must be string-based.  That means you can't create a `List` component using FAST and pass it an array.  However, you could do this with a combination of Blazor and FAST... use FAST UI elements and do all the virtualization with Blazor/javascript.

Anyways, I wonder if there's any merit to using the FAST Blazor implementation...

**Comment by limefrogyank** [comment]
Here's what components they have.  These are NOT the fluent version.  Unfortunately, you'll have to run your own version of this code and apply the fluent design to them.  But at least you can see what exists.

https://explore.fast.design/components/fast-accordion

**Reply by NPadrutt** [reply]
That was what kinda confused me. They moved the fluent part out to the fluent team. But the wrapper uses that fluent part again under the FAST name in t he fast repository.

**Reply by vnbaaij** [reply]
And by doing that they 'moved a lot of cheese'. 

For example, in Fluent UI (both React and ours), you use a `<PrimaryButton>` tag to create such a primary button. Now in FAST you need to create a `<FluentButton>` with an `Appearance="Appearance.Accent"` attribute. They used this mechanism with a buttonType attribute before in React and actually deprecated that a while back.

So, different mechanisms (attribute vs specific tag) and different terminology (Primary vs Appearance.Accent) to get to the same result. And as @limefrogyank already mentioned, the styling is not exactly the same. Not a good recipe for broad succesful addoption, I think.

**Comment by NPadrutt** [comment]
To your first point. I think the react versions currently are the "old" fluent and the northstar / teams react nugets are the newer and they work on kinda consolidating those - or at least I read something along those lines.

for what it's worth, I kinda struggled with the "normal" / non-blazor components of fast when I played around with them once. So definitely interesting to see if those components are more convenient to use.

**Comment by vnbaaij** [comment]
I wrote a blog about using FAST with Blazor earlier: https://baaijte.net/blog/fluent-blazor/. Think most of it still stands with the release of this package.
It's a nice starting point but I think we still have a lot more 'coverage' of the React library then FAST has (at the moment) and that you can use the native Blazor functionality for things like EventHandlers and bindings are a big plus for our implementation. Will be interesting to see where this is going though.

**Comment by limefrogyank** [comment]
From the reveal thread, looks like they are working towards visual parity with react and more components:
https://github.com/dotnet/aspnetcore/issues/11229#issuecomment-834683243

**Reply by vnbaaij** [reply]
I'm in contact with the people at Microsoft working on this (and plan to contribute to the package in the future). If there is anything you want me to relay to them, let me know.

**Reply by limefrogyank** [reply]
I'm using their FAST components to create a web component that is going to show up in a website written mostly with Perl.  It's nice to be able to encapsulate all of the javascript into a web component that you can load very easily into any web environment.  Far from finished, but:

![image](https://user-images.githubusercontent.com/7821384/124671238-0c592300-de6a-11eb-9a9a-97cba6be7fc4.png)
